proposals/lip-0047.md

```diff
@@ -498,53 +499,194 @@ This module does not define any specific logic for other modules.
 
 This module does not define any endpoint for off-chain services.
 
-### Protocol Logic During Block Lifecycle
+### Genesis Block Processing
 
-#### After Genesis Block Execution
+The following two steps are executed as part of the genesis block processing, see the “Update genesis block schema and processing” LIP for details. 
 
-After the genesis block `g` is executed, the following logic is executed:
+#### Genesis State Initialization
+
+To initialize the state of a genesis block `g`, the following logic is executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the bytes included in the `data` property of a deserialized entry in the block assets for the PoA module and let `assetPoA` be the deserialization of `genesisBlockAssetBytes` according to the `genesisPoAStoreSchema` schema, then: 
+
+* Checks for the `assetPoA.validators` (note that these checks are equivalent to those for the verification of the Authority Registration command): 
+  * Check that the `address` properties of all entries in the `assetPoA.validators` array are pairwise distinct.
+  * Check that the `name` property of every entry in the `assetPoA.validators` array contains only characters from the set `[a-z0-9!@$&_.]`, is not empty and is at most `MAX_LENGTH_NAME` characters long.
+  * Check that the `name` property of all entries in the `assetPoA.validators` array are pairwise distinct. 
+  * Check that the `generatorKey` property of every entry in the `assetPoA.validators` array has length 32.
+  * Check that the `blsKey` property of every entry in the `assetPoA.validators` array has length 48.
+  * Check that the `proofOfPossession` property of every entry in the `assetPoA.validators` array has length 96.
+
+* Checks for the `assetPoA.snapshotSubstore`: 
```

Comment:
We do not check the number of entries of the array as for the authority update command.

Reply:
Regarding the comment above.

>  I think in this case it is a bit dangerous to refer to that logic since for the update authority command verification the entries in the validator substore already exist and we do not need to check for uniqueness etc.

Also in the authority update command we check address uniqueness. I think all checks are the same or which one specifically is different?

Reply:
Yes, the checks are equivalent but the command one is against the existing state store and here it is against the content of another property in the assets data. That is why I would prefer not to just point out to those checks. I have added a similar explanatory sentence as I did for checks for the `assetPoA.validators` though.

Good point regarding the number of entries, added.